flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -1727,6 +1731,99 @@ protected void cancelTask() {
+                            sleepTimeOutsideMail,
+                            ioMetricGroup.getIdleTimeMsPerSecond());
+            // Make sure WaitingThread is started after Task starts processing.
+            executor.submit(
+                    waitingThread::start,
+                    "Start WaitingThread after Task starts processing input.");
+
+            long startTs = System.currentTimeMillis();
+            throughputCalculator.calculateThroughput();
+            throughputCalculator.incomingDataSize(incomingDataSize);
+            task.invoke();
+            long resultThroughput = throughputCalculator.calculateThroughput();
+            long totalDuration = System.currentTimeMillis() - startTs;
+
+            assertThat(
+                    resultThroughput,
+                    greaterThanOrEqualTo(
+                            incomingDataSize * 1000 / (totalDuration - sleepTimeOutsideMail)));
+        } finally {
+            if (waitingThread != null) {
+                waitingThread.join();
+            }
+        }
+    }
+
+    @Test
+    public void testThroughputSchedulerStartsOnInvoke() throws Exception {
+        // given: Throughput meter which finishes the task when calculateThroughput was called.
+        UnAvailableTestInputProcessor inputProcessor = new UnAvailableTestInputProcessor();
+        AtomicInteger desirableNumberOfCalculation = new AtomicInteger(2);
+        try (MockEnvironment mockEnvironment =
+                new MockEnvironmentBuilder()
+                        .setTaskConfiguration(
+                                new Configuration().set(IN_FLIGHT_DATA_ADJUSTMENT_PERIOD, 1))
+                        .setThroughputMeter(
+                                new ThroughputCalculator(SystemClock.getInstance(), 10) {
+                                    @Override
+                                    public long calculateThroughput() {
+                                        if (desirableNumberOfCalculation.decrementAndGet() == 0) {
+                                            inputProcessor
+                                                    .availabilityProvider
+                                                    .getUnavailableToResetAvailable()
+                                                    .complete(null);
+                                        }
+                                        return super.calculateThroughput();
+                                    }
+                                })
+                        .build()) {
+
+            StreamTask task =
+                    new MockStreamTaskBuilder(mockEnvironment)
+                            .setStreamInputProcessor(inputProcessor)
+                            .build();
+
+            // when: Starting the task.
+            task.invoke();
+
+            // then: The task successfully finishes because throughput calculation was called from
+            // scheduler which leads to the finish of the task
+        }
+    }
```

Comment:
What is the purpose of this test? Isn't it superseded by/subset of the `testMailboxNotifyThroughputMeterAboutIdleTime`?

Reply:
This test checks the fact that throughput calculation is started automatically(just to be sure that the scheduler is configured). But the calculation inside of `testMailboxNotifyThroughputMeterAboutIdleTime` is triggered manually in order to have more predictable value.

Reply:
Thanks for the explanation. Makes sense. Can you maybe copy paste the above explanation in the java doc of this test?